corebuild/integration/ILLink.Tasks/ILLink.Tasks.targets

```diff
@@ -328,21 +330,26 @@
        to work on the publish output. -->
   <Target Name="_ComputeLinkerRootAssemblies"
           DependsOnTargets="_ComputeManagedResolvedAssembliesToPublish;_ComputePlatformLibraries;_CheckSystemPrivateCorelibEmbeddedRoots">
-    <!-- By default, roots are everything minus the framework
-         assemblies. This doesn't include the intermediate
-         assembly, because we root it separately using an xml file,
-         which lets us explicitly root everything. 
+    <!-- If RootAllApplicationAssemblies is true, roots are everything minus the framework
+         assemblies. This doesn't include the intermediate assembly,
+         because we root it separately using an xml file,
+         which lets us explicitly root everything.
+
+         If RootAllApplicationAssemblies is false, only intermediate assembly's Main method is rooted.
+
          System.Private.CoreLib is rooted unless it has an embedded
          xml descriptor file. -->
     <ItemGroup>
       <_LinkerRootAssemblies Include="@(_ManagedResolvedAssembliesToPublish->'%(Filename)')" />
       <_LinkerRootAssemblies Remove="@(PlatformLibraries->'%(Filename)')" />
       <_LinkerRootAssemblies Remove="System.Private.CoreLib" Condition=" '$(_SPCHasEmbeddedRootDescriptor)' == 'true' "/>
-      <LinkerRootAssemblies Include="@(_LinkerRootAssemblies)" />
+      <LinkerRootAssemblies Include="@(_LinkerRootAssemblies)" Condition=" '$(RootAllApplicationAssemblies)' == 'true' " />
+
+      <LinkerRootAssemblies Include="@(IntermediateAssembly)" Condition=" '$(RootAllApplicationAssemblies)' == 'false' " />
+      <LinkerRootAssemblies Include="System.Private.CoreLib" Condition=" '$(RootAllApplicationAssemblies)' == 'false' And '$(_SPCHasEmbeddedRootDescriptor)' == 'false' "/>
     </ItemGroup>
```

Comment:
I think we should only compute `_LinkerRootAssemblies` when rooting all app assemblies, by doing something like this:
```
<ItemGroup Condition=" '$(RootAllApplicationAssemblies)' == 'true' ">
  <_LinkerRootAssemblies Include="@(_ManagedResolvedAssembliesToPublish->'%(Filename)')" />
  <_LinkerRootAssemblies Remove="@(PlatformLibraries->'%(Filename)')" />
  <_LinkerRootAssemblies Remove="System.Private.CoreLib" Condition=" '$(_SPCHasEmbeddedRootDescriptor)' == 'true' "/>
  <LinkerRootAssemblies Include="@(_LinkerRootAssemblies)" />
</ItemGroup>

<ItemGroup Condition=" '$(RootAllApplicationAssemblies)' != 'true' ">
  <LinkerRootAssemblies Include="@(IntermediateAssembly)" />
  <LinkerRootAssemblies Include="System.Private.CoreLib" Condition=" '$(_SPCHasEmbeddedRootDescriptor)' == 'false' "/>
</ItemGroup>
```

(Separating `_LinkerRootAssemblies` from `LinkerRootAssemblies` is useful in the first case because it makes it possible to manually root a platform assembly or SPC.

Reply:
@sbomer I made the change, PTAL.